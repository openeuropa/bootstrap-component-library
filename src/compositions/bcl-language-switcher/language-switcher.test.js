import { renderTwigFileAsNode } from "@openeuropa/bcl-test-utils";

import demoData from "@openeuropa/bcl-language-switcher/data";

const template = "@oe-bcl/bcl-language-switcher/language-switcher.html.twig";
const render = (params) => renderTwigFileAsNode(template, params);

describe("OE - Language switcher", () => {
  test("renders correctly", () => {
    expect.assertions(1);

    return expect(render(demoData)).resolves.toMatchSnapshot();
  });
});
