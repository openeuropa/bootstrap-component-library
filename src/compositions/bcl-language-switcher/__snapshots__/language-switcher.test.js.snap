// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OE - Language switcher renders correctly 1`] = `
<jest>
  <div
    class="alert alert-warning d-flex align-items-center alert-dismissible fade show text-dark"
    role="alert"
  >
    <svg
      class="flex-shrink-0 me-3 mt-1 align-self-start text-warning bi icon--s"
    >
      <use
        xlink:href="/icons.svg#exclamation-triangle-fill"
      />
    </svg>
    <div
      class="alert-content flex-grow-1"
    >
      This page is not available in [language]
    </div>
    <button
      aria-label="Close"
      class="btn-close"
      data-bs-dismiss="alert"
      type="button"
    />
  </div>
  <button
    aria-controls="dropdown-languages"
    aria-expanded="false"
    autocomplete="off"
    class="btn btn-outline-primary btn-md"
    data-bs-target="#dropdown-languages"
    data-bs-toggle="collapse"
    type="button"
  >
    Choose another language
    <svg
      class="ms-2-5 bi icon--fluid"
    >
      <use
        xlink:href="/icons.svg#caret-down-fill"
      />
    </svg>
  </button>
  <div
    class="collapse mt-3"
    id="dropdown-languages"
  >
    <div
      class="d-md-grid"
      style="grid-auto-flow: column; grid-template-rows: repeat(4, 1fr)"
    >
      <a
        class="btn btn-link d-block ps-0 text-start"
        href="#"
      >
        български
      </a>
      <a
        class="btn btn-link d-block ps-0 text-start"
        href="#"
      >
        čeština
      </a>
      <a
        class="btn btn-link d-block ps-0 text-start"
        href="#"
      >
        dansk
      </a>
      <a
        class="btn btn-link d-block ps-0 text-start"
        href="#"
      >
        Deutsch
      </a>
      <a
        class="btn btn-link d-block ps-0 text-start"
        href="#"
      >
        eesti
      </a>
      <a
        class="btn btn-link d-block ps-0 text-start"
        href="#"
      >
        ελληνικά
      </a>
      <a
        class="btn btn-link d-block ps-0 text-start"
        href="#"
      >
        <svg
          class="me-2 bi icon--xs"
        >
          <use
            xlink:href="/icons.svg#check-lg"
          />
        </svg>
        English
      </a>
      <a
        class="btn btn-link d-block ps-0 text-start"
        href="#"
      >
        français
      </a>
      <a
        class="btn btn-link d-block ps-0 text-start"
        href="#"
      >
        Gaeilge
      </a>
      <a
        class="btn btn-link d-block ps-0 text-start"
        href="#"
      >
        hrvatski
      </a>
      <a
        class="btn btn-link d-block ps-0 text-start"
        href="#"
      >
        italiano
      </a>
      <a
        class="btn btn-link d-block ps-0 text-start"
        href="#"
      >
        latviešu
      </a>
      <a
        class="btn btn-link d-block ps-0 text-start"
        href="#"
      >
        lietuvių
      </a>
      <a
        class="btn btn-link d-block ps-0 text-start"
        href="#"
      >
        magyar
      </a>
      <a
        class="btn btn-link d-block ps-0 text-start"
        href="#"
      >
        Malti
      </a>
      <a
        class="btn btn-link d-block ps-0 text-start"
        href="#"
      >
        Nederlands
      </a>
      <a
        class="btn btn-link d-block ps-0 text-start"
        href="#"
      >
        polski
      </a>
      <a
        class="btn btn-link d-block ps-0 text-start"
        href="#"
      >
        português
      </a>
      <a
        class="btn btn-link d-block ps-0 text-start"
        href="#"
      >
        română
      </a>
      <a
        class="btn btn-link d-block ps-0 text-start"
        href="#"
      >
        slovenčina
      </a>
      <a
        class="btn btn-link d-block ps-0 text-start"
        href="#"
      >
        slovenščina
      </a>
      <a
        class="btn btn-link d-block ps-0 text-start"
        href="#"
      >
        suomi
      </a>
      <a
        class="btn btn-link d-block ps-0 text-start"
        href="#"
      >
        svenska
      </a>
    </div>
  </div>
</jest>
`;
